web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -722,10 +755,18 @@ public void setViewport(double firstRow, double lastRow, @JsOptional JsArray<Col
             @JsOptional Double updateInterval) {
         this.firstRow = firstRow;
         this.lastRow = lastRow;
-        this.columns = columns != null ? Js.uncheckedCast(columns.slice()) : baseTable.getColumns();
+        this.columns = columns != null ? Js.uncheckedCast(columns.slice()) : visibleColumns;
```

Comment:
May want to freeze the columns.slice() here for consistency; visibleColumns is already frozen. Doesn't get passed to UI though so doesn't really matter.